tests/frontier/opcodes/test_calldatacopy.py

```diff
@@ -0,0 +1,255 @@
+            b"\x04",
+            Account(storage={0x00: 0x00}),
+            Account(storage={0x00: 0x00}),
+        ),
+        (
+            (
+                Op.PUSH1[0x9]
+                + Op.PUSH32[0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFA]
+                + Op.PUSH1[0x0]
+                + Op.CALLDATACOPY
+                + Op.PUSH1[0x0]
+                + Op.MLOAD
+                + Op.PUSH1[0x0]
+                + Op.SSTORE
+                + Op.MSIZE
+                + Op.PUSH1[0x0]
+                + Op.RETURN
+                + Op.STOP
+            ),
+            b"\x05",
+            Account(storage={0x00: 0x00}),
+            Account(storage={0x00: 0x00}),
+        ),
+        (
+            (
+                Op.PUSH1[0x1]
+                + Op.PUSH1[0x1]
+                + Op.SSTORE
+                + Op.PUSH1[0x1]
+                + Op.PUSH1[0x2]
+                + Op.CALLDATACOPY
+            ),
+            b"\x10",
+            Account(storage={0x01: 0x00}),
+            None,
+        ),
+        (
+            (
+                Op.PUSH1[0x5]
+                + Op.JUMP
+                + Op.JUMPDEST
+                + Op.STOP
+                + Op.JUMPDEST
+                + Op.PUSH1[0x42]
+                + Op.PUSH1[0x1F]
+                + Op.MSTORE8
+                + Op.PUSH2[0x103]
+                + Op.PUSH1[0x0]
+                + Op.PUSH1[0x1F]
+                + Op.CALLDATACOPY
+                + Op.PUSH1[0x0]
+                + Op.MLOAD
+                + Op.DUP1
+                + Op.PUSH1[0x60]
+                + Op.EQ
+                + Op.PUSH1[0x3]
+                + Op.JUMPI
+                + Op.PUSH5[0xBADC0FFEE]
+                + Op.PUSH1[0xFF]
+                + Op.SSTORE
+            ),
+            b"\x11",
+            Account(storage={0xFF: 0xBADC0FFEE}),
+            None,
+        ),
+    ],
+    ids=[
+        "cdc 0 1 2",
+        "cdc 0 1 1",
+        "cdc 0 1 0",
+        "cdc 0 0 0",
+        "cdc 0 neg6 ff",
+        "cdc 0 neg6 9",
+        "underflow",
+        "sec",
+    ],
+)
+def test_calldatacopy(
+    state_test: StateTestFiller,
+    code: Bytecode,
+    tx_data: bytes,
+    pre: Alloc,
+    code_address_storage: Account,
+    to_address_storage: Account,
+):
+    """
+    Test `CALLDATACOPY` opcode.
+
+    Based on https://github.com/ethereum/tests/blob/develop/src/GeneralStateTestsFiller/VMTests/vmTests/calldatacopyFiller.yml
+    """
+    code_address = pre.deploy_contract(code)
+    to = pre.deploy_contract(
+        code=(
+            Op.PUSH17[0x1234567890ABCDEF01234567890ABCDEF0]
+            + Op.PUSH1[0x0]
+            + Op.MSTORE
+            + Op.PUSH1[0x40]
+            + Op.PUSH1[0x20]
+            + Op.PUSH1[0x10]
+            + Op.PUSH1[0xF]
+            + Op.PUSH1[0x0]
+            + Op.PUSH1[0x4]
+            + Op.CALLDATALOAD
+            + Op.PUSH2[0x1000]
+            + Op.ADD
+            + Op.PUSH3[0xFFFFFF]
+            + Op.CALL
```

Comment:
I think this and most if not all lines above can be rewritten using parentheses like `Op.CALL(0xFFFFFF, Op.ADD( ...`.

Reply:
+1